x-pack/test/api_integration/deployment_agnostic/README.md

```diff
@@ -0,0 +1,188 @@
+
+  describe('compression', () => {
+    before(async () => {
+      roleAuthc = await samlAuth.createM2mApiKeyWithRoleScope('admin');
+      internalHeaders = samlAuth.getInternalRequestHeader();
+    });
+    after(async () => {
+      await samlAuth.invalidateM2mApiKeyWithRoleScope(roleAuthc);
+    });
+    describe('against an application page', () => {
+      it(`uses compression when there isn't a referer`, async () => {
+        const response = await supertestWithoutAuth
+          .get('/app/kibana')
+          .set('accept-encoding', 'gzip')
+          .set(internalHeaders)
+          .set(roleAuthc.apiKeyHeader);
+        expect(response.header).to.have.property('content-encoding', 'gzip');
+      });
+    });
+  });
+}
+```
+Load all test files in `index.ts` under the same folder.
+
+4. Add Tests Entry File and FTR Config File for **Stateful** Deployment
+
+Create `stateful.index.ts` tests entry file and load tests:
+
+```ts
+import { DeploymentAgnosticFtrProviderContext } from './ftr_provider_context';
+
+export default function ({ loadTestFile }: DeploymentAgnosticFtrProviderContext) {
+  describe('apis', () => {
+    loadTestFile(require.resolve('./apis/<my_api>'));
+  });
+}
+```
+
+Create `stateful.config.ts` and link tests entry file:
+
+```ts
+import { createStatefulTestConfig } from './../../api_integration/deployment_agnostic/default_configs/stateful.config.base';
+
+export default createStatefulTestConfig({
+  testFiles: [require.resolve('./stateful.index.ts')],
+  junit: {
+    reportName: 'Stateful - Deployment-agnostic API Integration Tests',
+  },
+  // extra arguments
+  esServerArgs: [],
+  kbnServerArgs: [],
+});
+```
+5. Add Tests Entry File and FTR Config File for Specific **Serverless** Project
+
+Example for Observability project:
+
+oblt.index.ts
+```ts
+import { DeploymentAgnosticFtrProviderContext } from './ftr_provider_context';
+
+export default function ({ loadTestFile }: DeploymentAgnosticFtrProviderContext) {
+  describe('Serverless Observability - Deployment-agnostic api integration tests', () => {
+    loadTestFile(require.resolve('./apis/<my_api>'));
+  });
+}
+```
+
+oblt.serverless.config.ts
+```ts
+import { createServerlessTestConfig } from './../../api_integration/deployment_agnostic/default_configs/serverless.config.base';
+
+export default createServerlessTestConfig({
+  serverlessProject: 'oblt',
+  testFiles: [require.resolve('./oblt.index.ts')],
+  junit: {
+    reportName: 'Serverless Observability - Deployment-agnostic API Integration Tests',
+  },
+  // include settings from project controller
+  // https://github.com/elastic/project-controller/blob/main/internal/project/observability/config/elasticsearch.yml
+  esServerArgs: ['xpack.ml.dfa.enabled=false'],
+  kbnServerArgs: [
+    // defined in MKI control plane
+    '--xpack.uptime.service.manifestUrl=mockDevUrl',
+    // useful for testing (also enabled in MKI QA)
+    '--coreApp.allowDynamicConfigOverrides=true',
```

Comment:
Are there any limitations for remote cloud/project deployments on what can be configured?  Do you think it's worth mentioning in the readme?

Reply:
Good point. Putting custom arguments in config often leads to tests fail on Cloud/MKI because people may think FTR provision these arguments into the deployment.

I will explicitly mention that any server args changes in FTR config must be verified to be working on Cloud/MKI by default.

Reply:
Work for me